### DISCUSSION TITLE
Steam loses connection, is this normal? Is it possible to make the connection more stable?

### DISCUSSION THREAD

**Discussion by MisterGibs** [SteamRE/SteamKit#1339] [General]
![12334311111](https://github.com/SteamRE/SteamKit/assets/140836134/b2c296ec-fb8f-466e-bc71-948a2453078e)
Steam loses connection, is this normal? Is it possible to make the connection more stable?

**Comment by MisterGibs** [comment]
I note that the Internet on the host is always very stable and reliable. But still the connection is lost after 12-48 hours.

**Comment by DoubleBullet** [comment]


**Reply by Christian-Stieber** [reply]
> I have no idea how I would reconnect using the new guard mobile login requirements.
> Guard data isn't stored unless it is email auth code. And Sentry file is depreciated...

The sentry file is not "deprecated", it's outright dead since Steam doesn't send it anymore. Unless they added it back in and declared it as deprecated, that is :-)

I believe SteamKit supports the new login system. In fact I'm pretty sure that was where I got the requisite flow from to implement the new login on my framework. Thanks guys for the reverse engineering :-)

Basically, the login runs in 2 phases now: first phase is sending credentials and doing 2FA, regardless of what kind of 2FA is configured for the account and selected by the client. The result of this is a login key ("refresh token"), which is then given to Steam for an actual login.

The thing is -- this login key has a rather long validity time, and can be used as often as you want. It is MEANT to be stored. My own Steam framework is not even storing the passwords with the account data anymore -- if and when it needs to do a new login, indicated by getting a "login expired" response when presenting the login key, it just asks the user for the password too.

Like here is the usual startup that I see:
```
...
*** [2024-06-12 13:43:44]: running client ***
*** [2024-06-12 13:43:45]: starting login with key (expires 2024-10-23 15:15:04)
*** [2024-06-12 13:43:45]: Login result: OK
...
```
So, for this account, this particular bot won't have to do another interactive login until October... maybe longer, if I can figure out how to refresh these keys, because I'm pretty sure my official Steam client hasn't asked me for a login in ages.
